src/System.Private.CoreLib/shared/System/TimeZoneInfo.Unix.cs

```diff
@@ -1285,19 +1283,20 @@ private static void TZif_ParseJulianDay(string date, out int month, out int day)
         /// <returns>
         /// true if the parsing succeeded; otherwise, false.
         /// </returns>
-        private static bool TZif_ParseMDateRule(string dateRule, out int month, out int week, out DayOfWeek dayOfWeek)
+        private static bool TZif_ParseMDateRule(ReadOnlySpan<char> dateRule, out int month, out int week, out DayOfWeek dayOfWeek)
         {
             if (dateRule[0] == 'M')
             {
-                int firstDotIndex = dateRule.IndexOf('.');
-                if (firstDotIndex > 0)
+                int monthWeekDotIndex = dateRule.IndexOf('.');
+                if (monthWeekDotIndex > 0)
                 {
-                    int secondDotIndex = dateRule.IndexOf('.', firstDotIndex + 1);
-                    if (secondDotIndex > 0)
+                    ReadOnlySpan<char> weekDaySpan = dateRule.Slice(monthWeekDotIndex + 1);
+                    int weekDayDotIndex = weekDaySpan.IndexOf('.');
```

Comment:
Just curious why you changed this into a Slice + IndexOf instead of just doing IndexOf with the starting index as was previously done?

Reply:
I couldn't find an overload to `ReadOnlySpan<>.IndexOf` that took a value and the starting index.  I assume we don't have it because the expectation is to `Slice` to get to where you want to start, and then `IndexOf` from there.  However, that is a bit disappointing here, because then I'd need to write:

```C#
int secondDotIndex = dateRule.Slice(firstDotIndex + 1).IndexOf('.') + firstDotIndex + 1;
```

In order to get the index of the 2nd dot w.r.t. the `dateRule` span.

So instead, I opted to create a new Span variable for the 2nd section, so I didn't have to do the extra math to get the index from `dateRule`.

Reply:
Hmm.  I thought we had one, but you're right.